tests/test_sql_mixins.py

```diff
@@ -0,0 +1,219 @@
+                                    statement_id='bcd',
+                                    status=StatementStatus(state=StatementState.SUCCEEDED))
+
+    mocker.patch('databricks.sdk.service.sql.StatementExecutionAPI.get_statement', wraps=_get_statement)
+
+    response = w.statement_execution.execute('abc', 'SELECT 2+2')
+
+    assert response.status.state == StatementState.SUCCEEDED
+    assert response.result.byte_count == 100500
+
+
+def test_execute_poll_timeouts_on_client(config, mocker):
+    w = WorkspaceClient(config=config)
+
+    mocker.patch('databricks.sdk.service.sql.StatementExecutionAPI.execute_statement',
+                 return_value=ExecuteStatementResponse(status=StatementStatus(state=StatementState.PENDING),
+                                                       statement_id='bcd',
+                                                       ))
+
+    mocker.patch('databricks.sdk.service.sql.StatementExecutionAPI.get_statement',
+                 return_value=GetStatementResponse(status=StatementStatus(state=StatementState.RUNNING),
+                                                   statement_id='bcd'))
+
+    cancel_execution = mocker.patch('databricks.sdk.service.sql.StatementExecutionAPI.cancel_execution')
+
+    with pytest.raises(TimeoutError):
+        w.statement_execution.execute('abc', 'SELECT 2+2', timeout=timedelta(seconds=1))
+
+    cancel_execution.assert_called_with('bcd')
+
+
+def test_fetch_all_no_chunks(config, mocker):
+    w = WorkspaceClient(config=config)
+
+    mocker.patch('databricks.sdk.service.sql.StatementExecutionAPI.execute_statement',
+                 return_value=ExecuteStatementResponse(
+                     status=StatementStatus(state=StatementState.SUCCEEDED),
+                     manifest=ResultManifest(schema=ResultSchema(columns=[
+                         ColumnInfo(name='id', type_name=ColumnInfoTypeName.INT),
+                         ColumnInfo(name='since', type_name=ColumnInfoTypeName.DATE),
+                         ColumnInfo(name='now', type_name=ColumnInfoTypeName.TIMESTAMP),
+                     ])),
+                     result=ResultData(external_links=[ExternalLink(external_link='https://singed-url')]),
+                     statement_id='bcd',
+                 ))
+
+    raw_response = mocker.Mock()
+    raw_response.json = lambda: [["1", "2023-09-01", "2023-09-01T13:21:53Z"],
+                                 ["2", "2023-09-01", "2023-09-01T13:21:53Z"]]
+
+    http_get = mocker.patch('requests.sessions.Session.get', return_value=raw_response)
+
+    rows = list(
+        w.statement_execution.iterate_rows(
+            'abc', 'SELECT id, CAST(NOW() AS DATE) AS since, NOW() AS now FROM range(2)'))
+
+    assert len(rows) == 2
+    assert rows[0].id == 1
+    assert isinstance(rows[0].since, datetime.date)
+    assert isinstance(rows[0].now, datetime.datetime)
+
+    http_get.assert_called_with('https://singed-url')
```

Comment:
Can we assert that we _never_ submit the authorization token when fetching any external link?

Reply:
will do